conn.go

```diff
@@ -953,6 +959,68 @@ func (c *Conn) ReadPartitions(topics ...string) (partitions []Partition, err err
 	return
 }
 
+// ReadTopics returns the list of all the topcs in the Kafka cluster.
+func (c *Conn) ReadTopics() (topics []Topic, err error) {
+	err = c.readOperation(
+		func(deadline time.Time, id int32) error {
+			return c.writeRequest(metadataRequest, v1, id, topicMetadataRequestV1{})
+		},
+		func(deadling time.Time, size int) error {
+			var res metadataResponseV1
+
+			if err := c.readResponse(size, &res); err != nil {
+				return err
+			}
+
+			brokers := make(map[int32]Broker, len(res.Brokers))
+			for _, b := range res.Brokers {
+				brokers[b.NodeID] = Broker{
+					Host: b.Host,
+					Port: int(b.Port),
+					ID:   int(b.NodeID),
+					Rack: b.Rack,
+				}
+			}
+
+			makeBrokers := func(ids ...int32) []Broker {
+				b := make([]Broker, len(ids))
+				for i, id := range ids {
+					b[i] = brokers[id]
+				}
+				return b
+			}
+
+			topics = make([]Topic, 0, len(res.Topics))
+			for _, t := range res.Topics {
+				if t.TopicErrorCode != 0 && (c.topic == "" || t.TopicName == c.topic) {
+					// We only report errors if they happened for the topic of
```

Comment:
This error handling seems a little weird to me.  When would we have this case and why would we want to return the empty partitions slice?  It feels like we're losing information and hiding things from the caller.